internal/twirptest/service_test.go

```diff
@@ -834,6 +835,60 @@ func TestPanickyApplication(t *testing.T) {
 	recorder.assertHookCalls(t, []hookCall{received, routed, errored, sent})
 }
 
+// If an application panics in its handler, the Error hooks should trigger and have access to the panic source.
+func TestPanicsTriggerErrorHooks(t *testing.T) {
+	errHookCalled := false
+	errHook := &twirp.ServerHooks{
+		Error: func(ctx context.Context, twerr twirp.Error) context.Context {
+			errHookCalled = true
+			// The error should have a .Cause containing the panic value for inspection
+			err := pkgerrors.Cause(twerr)
+			if err.Error() != "This Is Fine Meme" {
```

Comment:
Rather than testing the error string, you should be able to test whether `err == panicValue` directly.